### PR TITLE
Stop using np.{builtin}, and fix bugs due to the previous confusion

### PR DESCRIPTION
**Pull request by eric-wieser**
For `x` in `{int, bool, str, float, complex, object}`, `np.{x}` and `builtins.{x}` are one and the same:

```
>>> np.float is float
True
>>> np.bool is bool
True
# etc
```

Using the `np.{x}` version is deceptive as it suggests that there is a difference, when there is not. This change doesn't affect any external behaviour.

Similar to numpy/numpy#9517

### CODE REVIEW

File: lib/matplotlib/mlab.py [hunk 1]
```diff
@@ -3136,17 +3136,17 @@ def get_type(item, atype=int):
     def get_justify(colname, column, precision):
         ntype = column.dtype
 
-        if np.issubdtype(ntype, str) or np.issubdtype(ntype, bytes):
```
Comment (eric-wieser) [inline lib/matplotlib/mlab.py]: This didn't catch unicode on python 2, but did on python 3

File: lib/matplotlib/mlab.py [hunk 2]
```diff
@@ -3136,17 +3136,17 @@ def get_type(item, atype=int):
...
+        if np.issubdtype(ntype, np.character):
             fixed_width = int(ntype.str[2:])
             length = max(len(colname), fixed_width)
             return 0, length+padding, "%s"  # left justify
 
-        if np.issubdtype(ntype, np.int):
```
Comment (eric-wieser) [inline lib/matplotlib/mlab.py]: This was false for `ntype == np.uint8`

File: lib/matplotlib/mlab.py [hunk 3]
```diff
@@ -3136,17 +3136,17 @@ def get_type(item, atype=int):
...
+        if np.issubdtype(ntype, np.integer):
             length = max(len(colname),
                          np.max(list(map(len, list(map(str, column))))))
             return 1, length+padding, "%d"  # right justify
 
-        if np.issubdtype(ntype, np.float):
```
Comment (eric-wieser) [inline lib/matplotlib/mlab.py]: This worked correctly, but will raise a warning in numpy 1.14 if numpy/numpy#9505 is merged

**Comment by tacaswell** [comment]
Thanks @eric-wieser !

**Pull request merged**